colorpicker/colorpicker.qml

```diff
@@ -1,30 +1,62 @@
 //  A toy QML colorpicker control, by Ruslan Shestopalyuk
-import Qt 4.7
+import QtQuick 2.11
+import QtQuick.Layouts 1.11
 import "content"
-import "content/ColorUtils.js" as ColorUtils
 
 Rectangle {
     id: colorPicker
-    property color colorValue: ColorUtils.hsba(hueSlider.value, sbPicker.saturation,
+    property color colorValue: _hsla(hueSlider.value, sbPicker.saturation,
                                                sbPicker.brightness, alphaSlider.value)
-    width: 300; height: 200
+    property bool enableAlphaChannel: true
+    property bool enableDetails: true
+
+    signal colorChanged(color changedColor)
+
+    width: 320; height: 200
     color: "#3C3C3C"
-    Row {
-        anchors.fill: parent
+
+    RowLayout {
+        id: row
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.verticalCenter: parent.verticalCenterCenter
+        anchors.top: parent.top
         spacing: 3
 
         // saturation/brightness picker box
         SBPicker {
             id: sbPicker
-            hueColor : hueSlider.value
-            width: parent.height; height: parent.height
+            Layout.alignment: Qt.AlignLeft | Qt.AlignTop
+            hueColor: {
+                var v = 1.0-hueSlider.value
+                console.debug("v:"+v)
+
+                if(0.0 <= v && v < 0.16) {
+                    return Qt.rgba(1.0, 0.0, v/0.16, 1.0)
+                } else if(0.16 <= v && v < 0.33) {
+                    return Qt.rgba(1.0 - (v-0.16)/0.17, 0.0, 1.0, 1.0)
+                } else if(0.33 <= v && v < 0.5) {
+                    return Qt.rgba(0.0, ((v-0.33)/0.17), 1.0, 1.0)
+                } else if(0.5 <= v && v < 0.76) {
+                    return Qt.rgba(0.0, 1.0, 1.0 - (v-0.5)/0.26, 1.0)
+                } else if(0.76 <= v && v < 0.85) {
+                    return Qt.rgba((v-0.76)/0.09, 1.0, 0.0, 1.0)
+                } else if(0.85 <= v && v <= 1.0) {
+                    return Qt.rgba(1.0, 1.0 - (v-0.85)/0.15, 0.0, 1.0)
+                } else {
+                    console.log("something wrong!?")
```

Comment:
Could probably more precise here, like "hue value is outside of expected boundaries of [0, 1]", or something like that?

Reply:
Thank you for quick review. Sooner or later, i will change the message..
BTW, now I'm trying : https://github.com/rshest/qml-colorpicker/issues/2
If you have any idea or solution about the issue, let me know.